File: app/src/main/java/com/confessionsearch/release1/DeleteAsync.java
package com.confessionsearch.release1;

import android.os.AsyncTask;

public class DeleteAsync extends AsyncTask<Notes, Void, Void> {

    private NoteDao mNoteDao;

    public DeleteAsync(NoteDao dao) {
        mNoteDao = dao;
    }

    @Override
    protected Void doInBackground(Notes... notes) {
        mNoteDao.delete(notes);
        return null;
    }
    }

File: app/src/main/java/com/confessionsearch/release1/data/notes/DeleteAsync.kt
package com.confessionsearch.release1.data.notes

class DeleteAsync(private val mNoteDao: NoteDao) :
    CoroutinesAsyncTask<Notes?, Void?, Void?>("delete")
{
    override fun doInBackground(vararg notes: Notes?): Void?
    {
        mNoteDao.delete(*notes)
        return null
    }
}